src/Element_TEST.cc

```diff
@@ -72,6 +72,82 @@ TEST(Element, Required)
   ASSERT_EQ(elem.GetRequired(), "1");
 }
 
+/////////////////////////////////////////////////
+TEST(Element, SetExplicitlySetInFile)
+{
+  // The heirarchy in xml:
+  // <parent>
+  //   <elem>
+  //     <child>
+  //       <grandChild/>
+  //     </child>
+  //     <sibling/>
+  //     <sibling2/>
+  //   </elem>
+  //   <elem2/>
+  // </parent>
+  sdf::ElementPtr parent = std::make_shared<sdf::Element>();
+  sdf::ElementPtr elem = std::make_shared<sdf::Element>();
+  elem->SetParent(parent);
+  parent->InsertElement(elem);
+  sdf::ElementPtr elem2 = std::make_shared<sdf::Element>();
+  elem2->SetParent(parent);
+  parent->InsertElement(elem2);
+
+  EXPECT_TRUE(elem->GetExplicitlySetInFile());
+
+  elem->SetExplicitlySetInFile(false);
+
+  EXPECT_FALSE(elem->GetExplicitlySetInFile());
+
+  elem->SetExplicitlySetInFile(true);
+
+  EXPECT_TRUE(elem->GetExplicitlySetInFile());
+
+  // the childs and siblings of the element should all be
+  // set to the same value when using this function
+  sdf::ElementPtr child = std::make_shared<sdf::Element>();
+  child->SetParent(elem);
+  elem->InsertElement(child);
+
+  sdf::ElementPtr sibling = std::make_shared<sdf::Element>();
+  sibling->SetParent(elem);
+  elem->InsertElement(sibling);
+
+  sdf::ElementPtr sibling2 = std::make_shared<sdf::Element>();
+  sibling2->SetParent(elem);
+  elem->InsertElement(sibling2);
+
+  sdf::ElementPtr child2 = std::make_shared<sdf::Element>();
+  child2->SetParent(child);
+  child->InsertElement(child2);
```

Comment:
It looks like the hierarchy described in the XML snippet above is missing `child2`

Reply:
Good catch! fixed in: https://github.com/osrf/sdformat/pull/542/commits/4fcfa21dbdfb07e5cefdd885430ca026e7fc796e